scripts/irods/test/test_icp.py

```diff
@@ -218,3 +218,181 @@ def test_icp_with_apostrophe_logical_path__issue_5759(self):
+
+        original_content = 'goin down to cowtown'
+        new_content = 'gonna see the cow beneath the sea'
+
+        try:
+            # Make an object and replicate to some target resource...
+            self.user.assert_icommand(['istream', 'write', logical_path], 'STDOUT', input=original_content)
+            self.user.assert_icommand(['irepl', '-R', self.target_resource, logical_path])
+
+            self.assertTrue(lib.replica_exists_on_resource(self.user, logical_path, self.user.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.user, logical_path, self.target_resource))
+            self.assertFalse(lib.replica_exists_on_resource(self.user, logical_path, self.other_resource))
+
+            # Make a new object with different content and copy over the first object (the copy should succeed).
+            self.user.assert_icommand(['istream', 'write', other_logical_path], 'STDOUT', input=new_content)
+            self.user.assert_icommand(['icp', '-f', '-R', self.target_resource, other_logical_path, logical_path])
+
+            # Assert that the copy occurred and the existing replicas have been updated appropriately.
+            self.assertTrue(lib.replica_exists_on_resource(self.user, logical_path, self.user.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.user, logical_path, self.target_resource))
+            self.assertFalse(lib.replica_exists_on_resource(self.user, logical_path, self.other_resource))
+
+            self.assertEqual(
+                str(0), lib.get_replica_status_for_resource(self.user, logical_path, self.user.default_resource))
+            self.assertEqual(str(1), lib.get_replica_status_for_resource(self.user, logical_path, self.target_resource))
+
+            self.assertEqual(
+                original_content,
+                self.user.assert_icommand(
+                    ['istream', '-R', self.user.default_resource, 'read', logical_path], 'STDOUT')[1].strip())
+
+            self.assertEqual(
+                new_content,
+                self.user.assert_icommand(
+                    ['istream', '-R', self.target_resource, 'read', logical_path], 'STDOUT')[1].strip())
+
+        finally:
+            self.user.assert_icommand(['irm', '-f', logical_path])
+            self.user.assert_icommand(['irm', '-f', other_logical_path])
+
+    def test_overwrite_fails_when_target_resource_has_no_replica__issue_6497(self):
+        object_name = 'test_overwrite_fails_when_target_resource_has_no_replica__issue_6497'
+        other_object_name = 'test_overwrite_fails_when_target_resource_has_no_replica__issue_6497_other'
+        logical_path = os.path.join(self.user.session_collection, object_name)
+        other_logical_path = os.path.join(self.user.session_collection, other_object_name)
+
+        original_content = 'goin down to cowtown'
+        new_content = 'gonna see the cow beneath the sea'
```

Comment:
duplicate poetry alert - need different words here.

Reply:
Right-o

Reply:
Done. Ran out of TMBG lyrics that live in my head :)